src/index.ts

```diff
@@ -45,9 +45,39 @@ function notUndefined<T>(x: T | undefined): x is T {
   return x !== undefined;
 }
 
-export async function dump(n: readonly SceneNode[]): Promise<F.DumpedFigma> {
+// Returns true if n is a visible SceneNode or
+// not a SceneNode (e.g. a string, number, etc.)
+function isVisible(n: any) {
+  if (typeof n !== "object") {
+    return true;
+  }
+
+  if (
+    !("visible" in n) ||
+    typeof n.visible !== "boolean" ||
+    !("opacity" in n) ||
+    typeof n.opacity !== "number" ||
+    !("removed" in n) ||
+    typeof n.removed !== "boolean"
+  ) {
+    return true;
+  }
+
+  return n.visible && n.opacity > 0 && !n.removed;
+}
```

Comment:
is `> 0` safe? or should it be `> 0.001` or something?
I'm also concerned that this may have unintended consequence on eg fills or other places where we process an object that is not a layer?

Reply:
- > 0 works in my tests but can change it to 0.001 to be sure.
- I considered that but it shouldn't respond to fills or other objects because this combo of keys (visible, opacity, removed) is unique to SceneNodes. Agree there's a risk though. The alternative is checking the type, but then we likely have to check for 10+ types. Are there any other ways?

Reply:
Incorporated 0.001 comment.